loom/engine/bindings/loom/lmSQLite.cpp

```diff
@@ -0,0 +1,705 @@
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindBytes for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;
+    }
+
+    int step()
+    {
+        int result = sqlite3_step(statementHandle); 
+        if(result == SQLITE_ERROR)
+        {
+            lmLogError(gSQLiteGroup, "Error calling step for database: %s", parentDB->getDBName());
+        }
+        return result;
+    }
+
+    bool stepAsync()
+    {
+        if(asyncStepInProgress)
+        {
+            lmLogError(gSQLiteGroup, "Attempting to run multiple stepAsync calls on the same statement for database: %s", parentDB->getDBName());
+            return false;
+        }
+
+        //set up the progress handler
+        sqlite3_progress_handler(parentDB->dbHandle, 
+                                    Statement::statementProgressVMIWait, 
+                                    Statement::stepAsyncProgress, 
+                                    this);
+
+        //start up the stepAsync thread
+        asyncStepInProgress = true;
+        loom_thread_start(Statement::stepAsyncBody, (void *)this);    
+        return true;
+    }
+
+    const char *columnName(int col)
+    {
+        return sqlite3_column_name(statementHandle, col);
+    }
+
+    int columnType(int col)
+    {
+        return sqlite3_column_type(statementHandle, col);
+    }
+
+    int columnInt(int col)
+    {
+        return sqlite3_column_int(statementHandle, col);
+    }
+
+    double columnDouble(int col)
+    {
+        return sqlite3_column_double(statementHandle, col);
+    }
+
+    const char* columnString(int col)
+    {
+        return (const char *)sqlite3_column_text(statementHandle, col);
+    }
+
+    utByteArray *columnBytes(int col)
+    {
+        int size;
+
+        //get the blob from the column
+        void *blob = (void *)sqlite3_column_blob(statementHandle, col);
+        if(blob == NULL)
+        {
+            return NULL;
+        }
+        size = sqlite3_column_bytes(statementHandle, col);
+
+        //valid blob so allocate byte array for it
+        utByteArray *bytes = new utByteArray();
+        bytes->allocateAndCopy(blob, size);     
+
+        return bytes;   
+    }
+
+    int reset()
+    {
+        int result = sqlite3_reset(statementHandle); 
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling reset for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+
+    int finalize()
+    {
+        int result = sqlite3_finalize(statementHandle); 
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling finalize for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+};
+
+
+
+
+//---Statement--- external function initialisation
```

Comment:
Just functions that can't be declared inside of the class {} due to static
or dependencies.
On 20 Feb 2015 2:19 AM, "Ben Garney" <notifications@github.com> wrote:

> In loom/engine/bindings/loom/lmSQLite.cpp
> <https://github.com/LoomSDK/LoomSDK/pull/261#discussion_r25040078>:
>
> > +
> > +    int finalize()
> > +    {
> > +        int result = sqlite3_finalize(statementHandle);
> > +        if(result != SQLITE_OK)
> > +        {
> > +            lmLogError(gSQLiteGroup, "Error calling finalize for database: %s with Result Code: %i", parentDB->getDBName(), result);
> > +        }
> > +        return result;
> > +    }
> > +};
> > +
> > +
> > +
> > +
> > +//---Statement--- external function initialisation
>
> What does "external function initialization" mean?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LoomSDK/LoomSDK/pull/261/files#r25040078>.
>